src/ECOS.jl

```diff
@@ -15,17 +15,17 @@ using Compat.SparseArrays
 using Compat.LinearAlgebra
 
 # Try to load the binary dependency
-if isfile(joinpath(dirname(@__FILE__),"..","deps","deps.jl"))
+if isfile(joinpath(dirname(@__DIR__), "deps", "deps.jl"))
     include("../deps/deps.jl")
 else
     error("ECOS not properly installed. Please run Pkg.build(\"ECOS\")")
 end
 
-# ver  [not exported]
-# Returns the version of ECOS in use
-function ver()
-    ver_ptr = ccall((:ECOS_ver, ECOS.ecos), Ptr{UInt8}, ())
-    return unsafe_string(ver_ptr)
+const ECOS_VERSION = let
+    ver_ptr = ccall((:ECOS_ver, ecos), Ptr{UInt8}, ())
```

Comment:
Does this code run before `__init__`? What kind of error is produced if the installation is broken?

Reply:
`__init__` is run every time the package is loaded. This code runs when the package code is evaluated, i.e. during precompilation. Above here we have a check that `deps.jl` exists, so the potential pitfall is if `deps.jl` is malformed somehow and doesn't contain the `ecos` definition, in which case this will result in an `UndefVarError`, though the same would happen with the previous approach.

Reply:
>  though the same would happen with the previous approach

Good point.

> This code runs when the package code is evaluated, i.e. during precompilation.

Is it possible for the ECOS library version to change without triggering a new precompilation? It would be weird for a stale version number to be kept around.

Reply:
Oh, that's an interesting question, I'm not sure. I think theoretically it could be possible that the package wouldn't be recompiled.